### PR TITLE
Add sites method and enhance examples

### PR DESCRIPTION
**Pull request by Paul-Isache**

### CODE REVIEW

File: example/example.js [hunk 1]
```diff
@@ -17,11 +17,26 @@ const opts = {
...
   }
 }
+const Stream = require('stream')
+
 fastify.register(fileManager, opts)
 
+let siteId = null
+// let driveId = null
+let folderId = null
+let fileId = null
```
Comment (mcollina) [inline example/example.js:27]: These should not be top level variables.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: example/example.js (hunk 1 above)
Comment (Paul-Isache) [reply example/example.js:27]: i used for testing to store the values.

Comment (mcollina) [reply example/example.js:27]: users of this module will copy-and-paste this example and create serious bugs. Maybe pass them as route params?

File: src/sharepoint.js [hunk 2]
```diff
@@ -105,14 +104,21 @@ function sharepoint (options) {
     return request(filePath, accessToken)
   }
 
+  const sites = async (domain) => {
+    const { accessToken } = await getAccessToken()
+
+    return request(`sites/${domain}`, accessToken)
+  }
```
Comment (mcollina) [inline src/sharepoint.js:111]: this needs a test

**Review by mcollina — CHANGES_REQUESTED** [review]

File: example/example.js (hunk 1 above)
Comment (Paul-Isache) [reply example/example.js:27]: okey.. i will use the route params to pass this informations

**Review by mcollina — APPROVED** [review]
lgtm

File: test.js [hunk 3]
```diff
@@ -44,6 +42,39 @@ test('set the sharepoint decorator and send invalid config', async (t) => {
   }
 })
 
+test('sites validation test', async (t) => {
```
Comment (mcollina) [inline test.js:45]: please add a `t.plan()` otherwise if the errors are not thrown the test will not fail.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: test.js (hunk 3 above)
Comment (Paul-Isache) [reply test.js:45]: okey

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**